lib/puppet_x/ldapquery.rb

```diff
@@ -87,9 +85,12 @@ def get_entries()
                     :time => 10) do |entry|
           entries << entry
         end
-        Puppet.debug(entries)
+        end_time = Time.now.to_i
+        Puppet.debug("Searching #{@base} for #{@attributes} using #{@filter} took #{end_time - start_time} seconds")
```

Comment:
I'd do:

~~~ ruby
start_time = Time.now
#…
end_time = Time.now
Puppet.debug("Searching #{@base} for #{@attributes} using #{@filter} took #{sprintf('%.3f', end_time - start_time)} seconds")
~~~

Or maybe just turn the entire thing into one giant `sprintf`.